clarity_ext/domain/container.py

```diff
@@ -81,26 +81,33 @@ class Container(DomainObjectMixin):
     CONTAINER_TYPE_96_WELLS_PLATE = 100
     CONTAINER_TYPE_STRIP_TUBE = 200
     CONTAINER_TYPE_TUBE = 300
+    CONTAINER_TYPE_PATTERNED_FLOW_CELL = 400
 
-    def __init__(self, mapping=None, container_type=None, resource=None):
+    def __init__(self, mapping=None, container_type=None, size=None):
         """
         :param mapping: A dictionary-like object containing mapping from well
         position to content. It can be non-complete.
+        :param container_type: One of the CONTAINER_TYPE_* constants
+        :param size: The size of the container. Object should support height and width
         :return:
         """
         self.mapping = mapping
         self.container_type = container_type
         self.id = None
         self.name = None
 
-        if self.container_type == self.CONTAINER_TYPE_96_WELLS_PLATE:
-            self.size = PlateSize(height=8, width=12)
-        elif self.container_type == self.CONTAINER_TYPE_STRIP_TUBE:
-            self.size = PlateSize(height=8, width=1)
-        elif self.container_type == self.CONTAINER_TYPE_TUBE:
-            self.size = PlateSize(height=1, width=1)
+        if size:
+            self.size = size
         else:
-            raise ValueError("Unknown plate type '{}'".format(self.container_type))
+            # TODO: Require the size to be sent in instead
+            if self.container_type == self.CONTAINER_TYPE_96_WELLS_PLATE:
+                self.size = PlateSize(height=8, width=12)
```

Comment:
Yes, I think this will be removed entirely in a later commit, as these actually come from the LIMS. However, I didn't want to increase the scope of the commit (the commit is only ensuring that these will only be set when a size is not provided.)